circuit_knitting_toolbox/circuit_cutting/cutting_evaluation.py

```diff
@@ -14,9 +14,113 @@
 from __future__ import annotations
 
 from collections.abc import Sequence
+from itertools import chain
+from multiprocessing.pool import ThreadPool
 
+import numpy as np
 from qiskit.circuit import QuantumCircuit, ClassicalRegister
-from ..utils.observable_grouping import CommutingObservableGroup
+from qiskit.quantum_info import PauliList
+from qiskit.primitives.base import BaseSampler
+from qiskit.result import QuasiDistribution
+
+from ..utils.observable_grouping import CommutingObservableGroup, ObservableCollection
+from ..utils.iteration import strict_zip
+from .qpd import (
+    QPDBasis,
+    SingleQubitQPDGate,
+    TwoQubitQPDGate,
+    generate_qpd_samples,
+    decompose_qpd_instructions,
+    WeightType,
+)
+from .cutting_decomposition import decompose_observables
+
+
+def execute_experiments(
+    circuits: QuantumCircuit | dict[str | int, QuantumCircuit],
+    observables: PauliList | dict[str | int, PauliList],
+    num_samples: int,
+    samplers: BaseSampler | dict[str | int, BaseSampler],
```

Comment:
We can actually handle `AerSampler` with `shots=None`. We just check for that specific case and switch to `ExactSampler`

Reply:
My first reaction was :laughing:.  But it's a reasonable idea.  If we're going to do this, we should give the user a warning as well.

Reply:
I implemented this, and I like the behavior.

Users passing in qiskit.primitives.Sampler or qiskit_aer.primitives.Sampler(shots=None) are explicitly asking for exact sub-expectation values, so it's not too far of a leap to default to a class which can do that for them.